doc/nrf/ug_app_dev.rst

```diff
@@ -0,0 +1,77 @@
+.. _ncs-app-dev:
+
+Application Development
+#######################
+
+This page describes the fundamentals about developing a new application based
+on the |NCS|.
+
+Build and Configuration System
+******************************
+
+Zephyr's build and configuration system
+=======================================
+
+The |NCS| build and configuration system is based on Zephyr's, which in
+turn uses the following building blocks as a foundation:
+
+  * CMake, the cross-platform build system generator
+  * Kconfig, a powerful configuration system also used in the Linux kernel
+  * Device Tree, a hardware description language that is used to describe the
+    hardware that the |NCS| will run on
+
+Since the build and configuration system used by the |NCS| is provided by
+Zephyr, we refer to the original Zephyr documentation here in order to avoid
+duplication. The relevant sections of the Zephyr documentation that cover the
+different building blocks mentioned above can be found here:
+
+  * :ref:`zephyr:application` is a complete guide to application development
+    with Zephyr, including the build and configuration system
+  * :ref:`zephyr:cmake-details` describes in-depth the usage of CMake for
+    Zephyr-based applications
+  * :ref:`zephyr:application_kconfig` contains a guide for Kconfig usage in
+    applications
+  * :ref:`zephyr:application_dt` explains how to use Device Tree and its overlays to
+    customize an application's Device Tree
+
+|NCS| additions
+===============
+
+As mentioned in the previous sections, the build and configuration system in the
+|NCS| is inherited from Zephyr. The |NCS| does however add some additional
+functionality that is covered in this section, and that you should be aware of
+when you start writing your own applications based on this SDK:
+
+  * The |NCS| provides an additional :file:`boilerplate.cmake` that must be
+    included before Zephyr's in your app's :file:`CMakeLists.txt`::
+
+      include(path/to/nrf/cmake/boilerplate.cmake)
+      include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+
+    Since the |NCS| does not require or provide an environment variable
+    equivalent to :makevar:`ZEPHYR_BASE`, you can either create your own or
+    specify the full path in your own :file:`CMakeLists.txt`. In order to obtain
+    the path to the :file:`nrf` folder you can use west::
+
+      west list nrf -f {abspath}
```

Comment:
I meant something like this:

```
execute_process(COMMAND west list -f {abspath} nrf OUTPUT_VARIABLE NRF_BASE
  RESULT_VARIABLE west_list_err
  OUTPUT_STRIP_TRAILING_WHITESPACE)
if (west_list_err)
  message(FATAL_ERROR "cannot find nrf directory with west")
endif()

include(${NRF_BASE}/cmake/boilerplate.cmake)
```

This would work regardless of where their NCS directory is, so it's safer to check into version control.